java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -1011,4 +1019,182 @@ public void zcard(BaseClient client) {
                 assertThrows(ExecutionException.class, () -> client.zcard("foo").get());
         assertTrue(executionException.getCause() instanceof RequestException);
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void zrange_by_index(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        Map<String, Double> membersScores = Map.of("one", 1.0, "two", 2.0, "three", 3.0);
+        assertEquals(3, client.zadd(key, membersScores).get());
+
+        assertArrayEquals(
+                new String[] {"one", "two"}, client.zrange(key, new RangeByIndex(0, 1)).get());
+
+        assertEquals(
+                Map.of("one", 1.0, "two", 2.0, "three", 3.0),
+                client.zrangeWithScores(key, new RangeByIndex(0, -1)).get());
+
+        assertArrayEquals(
+                new String[] {"three", "two"}, client.zrange(key, new RangeByIndex(0, 1), true).get());
+
+        assertArrayEquals(new String[] {}, client.zrange(key, new RangeByIndex(3, 1), true).get());
+
+        assertTrue(client.zrangeWithScores(key, new RangeByIndex(3, 1)).get().isEmpty());
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void zrange_by_score(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        Map<String, Double> membersScores = Map.of("one", 1.0, "two", 2.0, "three", 3.0);
+        assertEquals(3, client.zadd(key, membersScores).get());
+
+        assertArrayEquals(
+                new String[] {"one", "two"},
+                client
+                        .zrange(
+                                key, new RangeByScore(InfScoreBound.NEGATIVE_INFINITY, new ScoreBoundary(3, false)))
+                        .get());
+
+        assertEquals(
+                Map.of("one", 1.0, "two", 2.0, "three", 3.0),
+                client
+                        .zrangeWithScores(
+                                key,
+                                new RangeByScore(InfScoreBound.NEGATIVE_INFINITY, InfScoreBound.POSITIVE_INFINITY))
```

Comment:
You can add imports and
```suggestion
                                new RangeByScore(NEGATIVE_INFINITY, POSITIVE_INFINITY))
```

Reply:
You do need InfScoreBound here as if you were to have a InfLexBound it would not work.